Not asking for sudo passwords for password-less sudo users any more

wk8 opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/release.md#issue-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Running hack/make-rules/test-e2e-node.sh or test/e2e_node/conformance/run_test.sh
with a password-less sudo user on a dev box right now requires to first
create a password for that user, and then type it every time one wants
to run these tests. This patch is fixing this by not asking for sudo
credentials if it seems the user can run any command without a password.

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

ddebroy commented:
/ok-to-test

michmike commented:
/lgtm

michmike commented:
cc: @spiffxp

michmike commented:
/assign @vishh

wk8 commented:
@vishh Could you please take a look? Thanks!

cblecker commented on hack/make-rules/test-e2e-node.sh:
```diff
@@ -154,10 +154,17 @@ if [ ${remote} = true ] ; then
+  # Refresh sudo credentials if needed
+  if ping -c 1 -q metadata.google.internal &> /dev/null; then
+    echo 'Running on CGE, not asking for sudo credentials'
+  elif sudo --non-interactive "$(which /bin/bash)" -c true 2> /dev/null; then
```
```suggestion
  elif sudo --non-interactive "$(which bash)" -c true 2> /dev/null; then
```

wk8 replied:
Not sure why I did this in the 1st place... Good catch, thanks! :)

cblecker commented on test/e2e_node/conformance/run_test.sh:
```diff
@@ -24,9 +24,17 @@
+# Refresh sudo credentials if needed
+if ping -c 1 -q metadata.google.internal &> /dev/null; then
+  echo 'Running on CGE, not asking for sudo credentials'
+elif sudo --non-interactive "$(which /bin/bash)" -c true 2> /dev/null; then
```
same here

wk8 replied:
👍

wk8 committed:
Not asking for sudo passwords for password-less sudo users any more

Running hack/make-rules/test-e2e-node.sh or test/e2e_node/conformance/run_test.sh
with a password-less sudo user on a dev box right now requires to first
create a password for that user, and then type it every time one wants
to run these tests. This patch is fixing this by not asking for sudo
credentials if it seems the user can run any command without a
password.

Signed-off-by: Jean Rouge <rougej+github@gmail.com>

wk8 commented:
@cblecker : thanks for the review, addressed your comments.

cblecker commented:
/lgtm
/approve

wk8 commented:
/test pull-kubernetes-conformance-image-test

wk8 commented:
/test pull-kubernetes-e2e-gce

wk8 commented:
/test pull-kubernetes-node-e2e

cblecker commented:
/retest

cblecker commented:
/retest
